improver/psychrometric_calculations/condensation_trails.py

```diff
@@ -367,3 +304,152 @@ def process(self, *cubes: Union[Cube, CubeList]) -> Cube:
+        """
+        Create new cubes containing the max or min heights
+        for persistent or non-persistent contrail formation.
+
+        Args:
+            formation_cube (iris.cube.Cube):
+                Categorical cube of shape (engine_factor, pressure_level, lat (optional), lon (optional))
+            height_cube (iris.cube.Cube):
+                Height cube of shape (pressure_level, lat (optional), lon (optional))
+            non_persistent_result (np.ndarray):
+                Extracted height data for non-persistent contrails.
+            persistent_result (np.ndarray):
+                Extracted height data for persistent contrails.
+            operation (str):
+                Either "max" or "min" indicating the type of extraction performed.
+
+        Returns:
+            non_persistent_height_cube, persistent_height_cube (tuple of iris.cube.Cube):
+                Cubes containing the extracted heights for non-persistent and persistent contrails.
+        """
+
+        template_cube = formation_cube.slices_over("pressure").next()
+        template_cube.remove_coord("pressure")
+        template_cube.attributes.pop("contrail_type", None)
+        template_cube.attributes.pop("contrail_type_meaning", None)
+
+        non_persistent_cube = template_cube.copy(
+            data=non_persistent_result,
+        )
+        non_persistent_cube.rename(f"{operation}_height_non_persistent_contrail")
+        non_persistent_cube.units = height_cube.units
+
+        persistent_cube = template_cube.copy(
+            data=persistent_result,
+        )
+        persistent_cube.rename(f"{operation}_height_persistent_contrail")
+        persistent_cube.units = height_cube.units
+
+        return non_persistent_cube, persistent_cube
+
+    def process(self, formation_cube: Cube, height_cube: Cube) -> Tuple[Cube, Cube]:
+        """
+        Main entry point for this class to extract the maximum or minimum height where contrail
+        formation is categorized as Non-persistent or Persistent.
+
+        Args:
+            formation_cube (iris.cube.Cube):
+                Categorical cube of shape (engine_factor, pressure_level, lat (optional), lon (optional))
+            height_cube (iris.cube.Cube):
+                Height cube of shape (pressure_level, lat (optional), lon (optional))
+
+        Returns:
+            non_persistent_height_cube, persistent_height_cube (tuple of iris.cube.Cube):
+                Each cube has dimensions (engine_factor, lat (optional), lon (optional))
+                and contains the extracted height values for the respective contrail category.
+        """
+
+        height_data = height_cube.data
+        try:
+            broadcast_height = np.broadcast_to(height_data, formation_cube.shape)
+        except ValueError as broadcast_error:
+            raise ValueError(
+                f"Cannot broadcast height data of shape {height_data.shape} to formation_cube shape {formation_cube.shape}"
```

Comment:
```suggestion
        try:
            broadcast_height = np.broadcast_to(height_cube.data, formation_cube.shape)
        except ValueError as broadcast_error:
            raise ValueError(
                f"Cannot broadcast height data of shape {height_cube.shape} to formation_cube shape {formation_cube.shape}"
```
Separate variable for data not required.